### DISCUSSION TITLE
Action button for bulk action

### DISCUSSION THREAD

**Discussion by ishah-twilio** [twilio-labs/paste#942] [Office hours] [answered]
Hello,

What should be recommended behavior for action button of data table for selected records?
Any guidance on what should be default state of Action button when no data record is selected in the table?

1) Should Action button be disabled till user selects at least one record from the data table? or
2) Should Action button be active and show user an error message when user clicks on it?

![Screen Shot 2020-11-16 at 11 15 45 AM](https://user-images.githubusercontent.com/69016180/99297511-35567600-27fd-11eb-9eba-94a7cf4e6a77.png)

![Screen Shot 2020-11-16 at 11 17 44 AM](https://user-images.githubusercontent.com/69016180/99297778-9120ff00-27fd-11eb-8fed-08e2e26cc3a0.png)

**Comment by kate-porter** [comment, marked as answer]
Hey @ishah-twilio - 

Our recommendation on this one would be to have the "actions" button enabled at all times, and show an error if the user clicks it before they have selected any SIMs. We recommend this route, as there are two issues with having the button disabled by default:
- there is no way to know _why_ the button is disabled or how to enable it, and
- because the button is _above_ the table, a user using assistive technology would have already moved past it by the time they selected a table row, so they would have no way of knowing that the button became enabled.

The error modal you've shown here works, though I would suggest updating the copy that specifies that the checkboxes are on the "left-hand side of each ICCID", as "left-hand side" doesn't mean anything to non-sighted users. You should always try to avoid directional cues, as outlined in the [content guidelines](https://paste.twilio.design/content/product-style-guide#be-inclusive-of-people-and-devices). Instead, try something like, "Select the Super SIMs that you would like to update, then try again."

**Reply by kate-porter** [reply]
Also, this is out of the scope of your question, but I'm curious how the term "Actions" performed in usability testing. Was it clear to users what clicking that button would do?

Our general recommendation for buttons is to start button text with a verb (see the last item in the [do/don't section on buttons](https://paste.twilio.design/components/button#when-to-use-a-button)), so it's clear to the user was clicking it will do. With the text "Actions", I would wonder if clicking it will execute an action or show me a list of actions, for example.

It looks like the modal that the button opens is titled "Update selected SIMs"—to mimic that, could the button text be "Update SIMs"? That also might make it more clear to the user that they need to select SIMs first before clicking the button.

**Reply by ishah-twilio** [reply]
Action button tested very well in the studies. User had clear understanding and design matched users expectations. 
Though, I have another round of studies in December and I will test this concept again.

Thank you.
